devicetypes/smartthings/zwave-rgbw-bulb.src/zwave-rgbw-bulb.groovy

```diff
@@ -0,0 +1,353 @@
+}
+
+private def zwToStColor(red, green, blue, warmWhite, coldWhite) {
+	def stColor = [:]
+	def colors = [red, green, blue]
+	def hexColor = "#" + colors.collect { Integer.toHexString(it).padLeft(2, "0") }.join("")
+	stColor["color"] = hexColor
+	def hsv = rgbToHSV(*colors)
+	stColor["hue"] = hsv.hue
+	stColor["saturation"] = hsv.saturation
+	stColor["colorTemperature"] = zwaveWhiteToTemp(warmWhite, coldWhite);
+	log.debug(stColor)
+	stColor
+}
+
+def updated() {
+	response(refresh())
+}
+
+def installed() {
+	log.debug "installed()..."
+	sendEvent(name: "checkInterval", value: 1860, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "0"])
+	sendEvent(name: "level", value: 100, unit: "%")
+	sendEvent(name: "colorTemperature", value: COLOR_TEMP_MIN)
+	sendEvent(name: "color", value: "#000000")
+	sendEvent(name: "hue", value: 0)
+	sendEvent(name: "saturation", value: 0)
+}
+
+def configure() {
+	commands([
+		// Set the dimming ramp rate
+		zwave.configurationV2.configurationSet(parameterNumber: 0x10, size: 1, scaledConfigurationValue: 5)
+	])
+}
+
+def parse(description) {
+	def result = null
+	if (description != "updated") {
+		def cmd = zwave.parse(description)
+		if (cmd) {
+			result = zwaveEvent(cmd)
+			log.debug("'$description' parsed to $result")
+		} else {
+			log.debug("Couldn't zwave.parse '$description'")
+		}
+	}
+	result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd) {
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd) {
+	unschedule(offlinePing)
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchcolorv3.SwitchColorReport cmd) {
+	log.debug "got SwitchColorReport: $cmd"
+	def result = []
+	if (state.staged != null) {
+		state.staged.each{ k, v -> result << createEvent(name: k, value: v) }
+	}
+	result
+}
+
+private dimmerEvents(physicalgraph.zwave.Command cmd) {
+	def value = (cmd.value ? "on" : "off")
+	def result = [createEvent(name: "switch", value: value, descriptionText: "$device.displayName was turned $value")]
+	if (cmd.value) {
+		result << createEvent(name: "level", value: cmd.value == 99 ? 100 : cmd.value , unit: "%")
+	}
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	def linkText = device.label ?: device.name
+	[linkText: linkText, descriptionText: "$linkText: $cmd", displayed: false]
+}
+
+def buildOffOnEvent(cmd){
+	[zwave.basicV1.basicSet(value: cmd), zwave.switchMultilevelV3.switchMultilevelGet()]
```

Comment:
I find it very strange to do a `basic set` and then do `switch multi level get`, I'd rather keep it uniform to one command class for both Set/Get. I would recommend using Switch Multi-Level command class.

Reply:
Got it.  Done.